### PR TITLE
Add setBinaryPayload() function to HTTP library

### LINKED ISSUE #4269: No method to set a binary data to http response
In https://github.com/ballerinalang/ballerina/tree/f141e7ca0fea01976a6dceec5f568b15026fdff6/modules/ballerina-http/src/main/java/org/ballerinalang/net/http/nativeimpl/response there is no SetBinaryPayload

### PR DESCRIPTION
**Pull request by pubudu91**
Fix #4269

### CODE REVIEW

**Comment by pubudu91** [comment]
@hasithaa Added test cases

File: modules/ballerina-core/src/main/java/org/ballerinalang/runtime/message/BlobDataSource.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    public void setValue(byte[] value) {
+        this.value = value;
+    }
+
+    @Override
+    public void serializeData() {
+        try {
+            this.outputStream.write(value);
+            this.outputStream.close();
```
Comment (hasithaa) [inline modules/ballerina-core/src/main/java/org/ballerinalang/runtime/message/BlobDataSource.java]: Usually, when you close a stream, it should have its own try-catch block. In this code, if the write fails, outputstream will not be closed.

**Comment by manuranga** [comment]
I build with this PR and I see an issue with content length not updating.

```ballerina
import ballerina.net.http;
@http:configuration {basePath:"/"}
service<http> echo {
    @http:resourceConfig {
        methods:["GET"],
        path:"/"
    }
    resource echo (http:Request req, http:Response resp) {
        resp.setStringPayload("sssssssssss");
        _ = resp.send();
    }
}
```


```
curl 'http://localhost:9090/' -vk
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 9090 (#0)
> GET / HTTP/1.1
> Host: localhost:9090
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Connection: Keep-Alive
< Content-Length: 0
< 
* Excess found in a non pipelined read: excess = 11 url = / (zero-length body)
* Connection #0 to host localhost left intact
```

**Comment by manuranga** [comment]
My bad, I was on a older version of parent.

**Comment by manuranga** [comment]
My bad, I was on a older version of parent.

**Pull request merged**